pkg/services/publicdashboards/service/service.go

```diff
@@ -238,6 +245,79 @@ func (pd *PublicDashboardServiceImpl) GetMetricRequest(ctx context.Context, dash
 	return metricReqDTO, nil
 }
 
+func (pd *PublicDashboardServiceImpl) GetAnnotations(ctx context.Context, reqDTO AnnotationsQueryDTO, accessToken string) ([]*annotations.ItemDTO, error) {
+	_, dash, err := pd.GetPublicDashboard(ctx, accessToken)
+	if err != nil {
+		return nil, err
+	}
+
+	type annotation struct {
+		Datasource struct {
+			DType string `json:"type"`
+			Uid   string `json:"uid"`
+		}
+		Enable bool   `json:"enable"`
+		Type   string `json:"type"`
+		Target struct {
+			Limit    int64    `json:"limit"`
+			MatchAny bool     `json:"matchAny"`
+			Tags     []string `json:"tags"`
+			Type     string   `json:"type"`
+		}
+	}
+
+	type annotationsDto struct {
+		Annotations struct {
+			List []annotation `json:"list"`
+		}
+	}
+
+	dto := annotationsDto{}
+	bytes, err := dash.Data.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(bytes, &dto)
+	if err != nil {
+		return nil, err
+	}
+
+	anonymousUser, err := pd.BuildAnonymousUser(ctx, dash)
+	if err != nil {
+		return nil, err
+	}
+
+	var results []*annotations.ItemDTO
```

Comment:
@sdboyer how close are we to use the cue generated go types for dashboard to loop through in this case Annotation queries. We loop and manipulate the dashboard model in many places in the core code would be a good place to start using the generated types.

Reply:
For this purpose, we're close, plausibly close enough that this PR should be refactored to use [generated types](https://github.com/grafana/grafana/blob/main/pkg/coremodel/dashboard/dashboard_gen.go#L324-L358).

Looks like the main issue is that there's no specification for [Target](https://github.com/grafana/grafana/blob/main/pkg/coremodel/dashboard/dashboard_gen.go#L882) written down yet. But if the fields given here are indeed what's expected to be there (even if it's not ALL of them), then there's no reason this PR couldn't just add them [to the schema](https://github.com/grafana/grafana/blob/main/pkg/coremodel/dashboard/coremodel.cue#L269) then re-run `make gen-cue`.

Reply:
Once this PR is functional, I'll come back and tackle this.